internal/processing/processing.go

```diff
@@ -59,15 +60,26 @@ func (f *Factory) CalculateRequiredState(api *gatewayv1beta1.APIRule, config *he
 	var res State
 	pathDuplicates := hasPathDuplicates(api.Spec.Rules)
```

Comment:
`pathDuplicates` suggests is a list of duplicates not a `bool`, could we refactor it to something more boolean like `hasPathDuplicates`?